storage/range.go

```diff
@@ -1035,6 +1058,47 @@ func (r *Range) handleSkippedIntents(args proto.Request, intents []proto.Intent)
 	}
 }
 
+type replicaCorruptionError struct {
+	error
+	msg       string
+	file      string
+	line      int
+	processed bool // for now, required for testing
+}
+
+func (rce *replicaCorruptionError) Error() string {
+	if rce == nil {
+		rce = newReplicaCorruptionError("nil", nil)
+	}
+	return fmt.Sprintf("replica corruption (processed=%t) @ %s:%d: %s: %s",
+		rce.processed, rce.file, rce.line, rce.msg, rce.error)
+}
+
+func newReplicaCorruptionError(msg string, err error) *replicaCorruptionError {
+	file, line := log.Caller(1)
+	return &replicaCorruptionError{
+		file:  file,
+		line:  line,
+		error: err,
+		msg:   msg,
+	}
+}
+
+// maybeSetCorrupt is a stand-in for proper handling of failing replicas. Such a
```

Comment:
I didn't want to get into actual handling yet for this PR, it's already getting a bit out of hand.